src/ossl/aes.rs

```diff
@@ -937,11 +937,19 @@ impl AesOperation {
                 /* Check the data length in CCM matches the provided data -- this is one-shot
                  * operation only */
                 if op.params.datalen != keydata.len() {
+                    zeromem(keydata.as_mut_slice());
                     return Err(CKR_MECHANISM_PARAM_INVALID)?;
                 }
             }
             _ => (),
         }
+
+        let needed_length = op.encryption_len(keydata.len(), true)?;
+        if output.len() < needed_length {
+            zeromem(keydata.as_mut_slice());
+            return Ok(needed_length);
+        }
```

Comment:
This needs to be split in 2 checks,
if output.len() is exactly 0 then returning Ok(needed_length) is correct, for any other value of the output's length see the comment on the rsa file

Reply:
My bad. Fixed now!